src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -190,6 +192,16 @@ CitusBeginScan(CustomScanState *node, EState *estate, int eflags)
 		CitusBeginModifyScan(node, estate, eflags);
 	}
 
+	Job *workerJob = scanState->distributedPlan->workerJob;
+	Const *distArgument = GetInForceDelegatedFuncExecution();
+	if (distArgument && !equal(distArgument, workerJob->partitionKeyValue))
```

Comment:
Maybe use datumIsEqual, not sure how equal(Const,Const) reacts to variations in fields like `location`, which is used to specify character in error messages, does not affect semantics.

Reply:
That's a valid point, I'm fine with using datumIsEqual(), apparently location is not compared?

/* Compare a parse location field (this is a no-op, per note above) */
#define COMPARE_LOCATION_FIELD(fldname) \
        ((void) 0)